src/templates/checkout/shipping.twig

```diff
@@ -11,18 +11,23 @@
 	{% set availableShippingMethodOptions = cart.availableShippingMethodOptions ?? null %}
 	{% set onlyMethod = null %}
 	{% if cart is defined %}
-		{# If there's only 1 shipping method available then set it on the cart #}
-		{% if cart.availableShippingMethodOptions|length == 1 %}
-			{% for handle, option in cart.availableShippingMethodOptions %}
-				{% if option.price == 0 %}
-					{% redirect(craft.fostercheckout.getPath('checkout') ~ '/payment') %}
-				{% else %}
+		{% if currentStore.requireShippingMethodSelectionAtCheckout == false %}
+			{# If there's 1 or more shipping method available then set it on the cart #}
+			{% if cart.availableShippingMethodOptions|length == 1 or cart.availableShippingMethodOptions|length > 1  %}
```

Comment:
`cart.availableShippingMethodOptions|length >= 1`

Reply:
Fixed in https://github.com/FosterCommerce/foster-checkout/pull/45